src/interfaces/interfaces.ts

```diff
@@ -46,7 +46,16 @@ namespace interfaces {
     prototype: T;
   }
 
-  export type ServiceIdentifier<T = unknown> = (string | symbol | Newable<T> | Abstract<T>);
+  export type IfAny<T, Y, N> = 0 extends (1 & T) ? Y : N;
+
+  export type PropertyServiceIdentifier = (string | symbol);
+  export type ServiceIdentifier<T = unknown> = (PropertyServiceIdentifier | Newable<T> | Abstract<T>);
+  export type BindingMap = Record<PropertyServiceIdentifier, any>;
```

Comment:
is there any way we can have this as `Record<PropertyServiceIdentifier, unknown>`; ? avoiding the `any` there i mean

Reply:
What's the benefit? I think `any` is more semantic here?

Reply:
no real benefit, keep consistency with most of the types since I tried really hard at some point to move away from most of `any`'s and have "proper" types

Reply:
I had a quick go, and it didn't immediately compile. I'm not personally sure it's worth investigating, because as I say the semantics aren't really "correct" (according to my limited understanding of `unknown` vs `any` 😅 ), since the values assigned in this `BindingMap` will be entirely consumer-generated, and they can legitimately set it to anything they want (ie it is semantically `any`, right...?).

If you feel strongly I can try to investigate, but the types don't really like it as-is:

```
src/container/container.ts:287:41 - error TS2345: Argument of type 'this' is not assignable to parameter of type 'Container<BindingMap, any>'.
  Type 'Container<T, P>' is not assignable to type 'Container<BindingMap, any>'.
    The types returned by 'bind(...)' are incompatible between these types.
      Type 'BindingToSyntax<ContainerBinding<T, K>>' is not assignable to type 'BindingToSyntax<B>'.
        Type 'ContainerBinding<T, K>' is not assignable to type 'B'.
          'ContainerBinding<T, K>' is assignable to the constraint of type 'B', but 'B' could be instantiated with a different subtype of constraint 'unknown'.
            Type 'ContainerBinding<T, string> | ContainerBinding<T, symbol>' is not assignable to type 'B'.
              'ContainerBinding<T, string> | ContainerBinding<T, symbol>' is assignable to the constraint of type 'B', but 'B' could be instantiated with a different subtype of constraint 'unknown'.
                Type 'ContainerBinding<T, string>' is not assignable to type 'B'.
                  'ContainerBinding<T, string>' is assignable to the constraint of type 'B', but 'B' could be instantiated with a different subtype of constraint 'unknown'.
                    Type 'T[keyof T & string]' is not assignable to type 'B'.
                      'T[keyof T & string]' is assignable to the constraint of type 'B', but 'B' could be instantiated with a different subtype of constraint 'unknown'.
                        Type 'T[string]' is not assignable to type 'B'.
                          'T[string]' is assignable to the constraint of type 'B', but 'B' could be instantiated with a different subtype of constraint 'unknown'.

287       const request = createMockRequest(this, serviceIdentifier, key, value);
```

Reply:
I don't really think you can go around this tbh. If you can that would be amazing. depends what @notaphplover thinks also